fenched_code_block.py

```diff
@@ -1,13 +1,90 @@
+import re
 from block import Block
 
+SUPPORTED_LANGUAGE = {
```

Comment:
[nit]: _SUPPORTED_LANGUAGE, and _LANGUAGE_MAPPING. 

Or if you want to reference those 2 structure out side, then ignore this comment